lib/vanagon/project/dsl.rb

```diff
@@ -171,6 +171,21 @@ def generate_archives(archive)
         @project.compiled_archive = archive
       end
 
+      # Sets the release for the project to the number of commits since the
+      # last tag. Requires that a git tag be present
+      # and reachable from the current commit in that repository.
+      #
+      def release_from_git
+        open_git = Git.open(File.expand_path("..", Vanagon::Driver.configdir))
+        last_tag = open_git.describe('HEAD', { :abbrev => 0 })
+        # log has to be given an arbitrarily large parameter so that
+        # it will fetch more than 30 commits (which is the default)
+        git_release = open_git.log(10_000).between(last_tag, 'HEAD').count
```

Comment:
I'm wondering if you need to be counting log entries, or if just parsing the second part of the git describe would be ok? If `git describe` returns `0.15.2-5-g332dbcc`, can you just split on '-' and take the second to last part as the number of commits? (I'm looking at the `version_from_git` method below)

Reply:
(Morgan doesn't need my validation, but I was going to leave a similar comment)

Reply:
(Yes, I stubbed out every single option to `git rev-list`, and followed the "support dash-case and snake_case" examples of the upstream Git library. For **reasons**.)